### DISCUSSION TITLE
Simulating migration in the final generation

### DISCUSSION THREAD

**Discussion by mshpak76** [MesserLab/SLiM#463] [General]
I am trying to model a scenario where two (or more) populations evolve without gene flow for N-1 generations followed by admixture where a fraction f of the individuals in population 3 are from population 2. My question is that if in generation N, immediately before writing to the output file, I have the SetMigrationRates function, will migration actually occur in this generation. Specifically, in my simulation, for the 2000th generation I have:

    2000 late() {
     m2mut = sim.mutationsOfType(m2);
     p3.setMigrationRates(p2,2e-1);
     freqs = sim.mutationFrequencies(p3, m2mut);
     sim.deregisterScriptBlock(self);
     writeFile("Terminal_Frequency.txt",asString(freqs*20000));
     sim.treeSeqOutput("ThreePopFix.trees");
     sim.simulationFinished();
   }

will this give the desired result? When I implemented it, there were no errors but the resulting allele frequencies suggested that there was no migration at all (i.e. the same result as I had without the setMigrationRates() function).

**Comment by bhaller** [comment]
Hi!  Questions like this are better for the slim-discuss mailing list, just so you know; but it is not a big deal.  :->  Many demographic changes, such as changes to migration rates, are "future requests"; they do not change the simulation state immediately, but rather, change what will happen the next time reproduction occurs.  So no, your code above would write out the same state either way; the new migration rate would not have any effect until the next reproduction phase in the tick cycle.
